connectors/sources/github.py

```diff
@@ -919,49 +1080,106 @@ def get_default_configuration(cls):
+                    owner=self.github_client.user
+                )
+                async for repo in self.github_client.get_user_repos():
+                    self.user_repos[repo["nameWithOwner"]] = repo
+                invalid_repos = list(
+                    set(configured_repos) - set(self.user_repos.keys())
+                )
 
-            for repo_name in self.github_client.repos:
-                if repo_name not in ["", None]:
-                    if "/" in repo_name:
-                        foreign_repos.append(repo_name)
-                    else:
-                        configured_repos.append(
-                            f"{self.github_client.user}/{repo_name}"
+                for repo_name in foreign_repos:
+                    try:
+                        self.foreign_repos[
+                            repo_name
+                        ] = await self.github_client.get_foreign_repo(
+                            repo_name=repo_name
                         )
-            async for repo in self.github_client.get_user_repos():
-                self.user_repos[repo["nameWithOwner"]] = repo
-            invalid_repos = list(set(configured_repos) - set(self.user_repos.keys()))
-
-            for repo_name in foreign_repos:
-                try:
-                    self.foreign_repos[
-                        repo_name
-                    ] = await self.github_client.get_foreign_repo(repo_name=repo_name)
-                except Exception:
-                    self._logger.debug(f"Detected invalid repository: {repo_name}.")
-                    invalid_repos.append(repo_name)
+                    except Exception:
+                        self._logger.debug(f"Detected invalid repository: {repo_name}.")
+                        invalid_repos.append(repo_name)
+            else:
+                foreign_repos, configured_repos = self.github_client.bifurcate_repos(
+                    owner=self.github_client.org_name
+                )
+                configured_repos.extend(foreign_repos)
+                async for repo in self.github_client.get_org_repos():
+                    self.org_repos[repo["nameWithOwner"]] = repo
+                invalid_repos = list(set(configured_repos) - set(self.org_repos.keys()))
             return invalid_repos
         except Exception as exception:
             self._logger.exception(
                 f"Error while checking for inaccessible repositories. Exception: {exception}."
             )
             raise
 
+    async def _user_access_control_doc(self, user):
+        user_id = user.get("id", "")
+        user_name = user.get("login", "")
+        user_email = user.get("email", "")
+
+        _prefixed_user_id = _prefix_user_id(user_id=user_id)
+        _prefixed_user_name = _prefix_username(user=user_name)
+        _prefixed_email = _prefix_email(email=user_email)
+        return {
+            "_id": user_id,
```

Comment:
What happens if you get multiple users with an empty id? Will they override each other or will new documents created with subsequent syncs, if ES generates new unique ids for users having an empty id?

Reply:
I don't think we get an empty ID for any document, as we have used the ID that was generated by GraphQL to identify the records. Only emails can be empty for some users. We always get a unique ID and username for all the users.